addons/GME/Scripts/Game/GME/Components/GME_AmbientFlybyComponent.c

```diff
@@ -40,45 +42,48 @@ class GME_AmbientFlybyComponent : ScriptComponent
-		compartmentManager.GetOnDoneSpawningDefaultOccupants().Insert(OnCrewSpawned);
-		compartmentManager.SpawnDefaultOccupants({ECompartmentType.PILOT,  ECompartmentType.TURRET});		
+		SCR_AIGroup crewGroup = GME_VehicleHelper.SpawnCrew(Vehicle.Cast(owner));
+		crewGroup.GME_GetOnDoneSpawningMembers().Insert(OnCrewSpawned);
+		
 		HelicopterControllerComponent ctrl = HelicopterControllerComponent.Cast(owner.FindComponent(HelicopterControllerComponent));
-		ctrl.StartEngine();
+		if (ctrl)
+			ctrl.StartEngine();
+		
+		IEntity child = owner.GetChildren();
+		while (child)
+		{
+			SCR_RotorDamageManagerComponent damageManager = SCR_RotorDamageManagerComponent.Cast(child.FindComponent(SCR_RotorDamageManagerComponent));
+			if (damageManager)
+			{
+				damageManager.GetOnDamageStateChanged().Insert(OnRotorDamageStateChanged);
+				m_aRotorDamageManagers.Insert(damageManager);
+			}
+			
+			child = child.GetSibling();
+		}
 	}
 	
 	//------------------------------------------------------------------------------------------------
-	protected void OnCrewSpawned(SCR_BaseCompartmentManagerComponent compartmentManager = null, array<IEntity> spawnedCharacters = null, bool wasCanceled = false)
+	protected void OnCrewSpawned(SCR_AIGroup group)
 	{
-		foreach (IEntity member : spawnedCharacters)
-		{
-			AIComponent aiComponent = AIComponent.Cast(member.FindComponent(AIComponent));
-			AIAgent agent = aiComponent.GetAIAgent();
-			m_aAgents.Insert(agent);
-			m_aAgents.Insert(agent.GetParentGroup());
-		}
+		group.GME_GetOnDoneSpawningMembers().Remove(OnCrewSpawned);
+		group.GetAgents(m_aAgents);
+		m_aAgents.Insert(group);
 		
 		foreach (AIAgent agent : m_aAgents)
 		{
 			agent.SetPermanentLOD(0);
 		}
 		
-		GetGame().GetCallqueue().CallLater(OnCrewSpawnedDelayed, 100, false, compartmentManager, spawnedCharacters, wasCanceled);
-		compartmentManager.GetOnDoneSpawningDefaultOccupants().Remove(OnCrewSpawned);
-	}
-	
-	//------------------------------------------------------------------------------------------------
-	protected void OnCrewSpawnedDelayed(SCR_BaseCompartmentManagerComponent compartmentManager = null, array<IEntity> spawnedCharacters = null, bool wasCanceled = false)
-	{
-		foreach (IEntity character : spawnedCharacters)
-		{
-			SCR_CompartmentAccessComponent compartmentAccess = 	SCR_CompartmentAccessComponent.Cast(character.FindComponent(SCR_CompartmentAccessComponent));
-			if (compartmentAccess.GetCompartment().GetType() == ECompartmentType.PILOT)
-			{
-				SCR_CharacterControllerComponent characterController = SCR_CharacterControllerComponent.Cast(character.FindComponent(SCR_CharacterControllerComponent));
-				characterController.m_OnLifeStateChanged.Insert(OnPilotLifeStateChanged);
-			}
-		}
+		SCR_ChimeraCharacter pilot = SCR_ChimeraCharacter.Cast(m_aAgents[0].GetControlledEntity());
+		if (!pilot)
+			return;
```

Comment:
What happens if we return here? Do we end up with a component and vehicle sitting there but not executing the rest of the functionality due to no pilot? 
Or does the module clean itself up, or does it never spawn the vehicle if we return this early?

Reply:
Oh it just happily continues with the reset of the simulation 😅 
I should rethink this. Maybe delete the helo when this happens.

Reply:
Might be one of the modules where an option in the module framework to register a "cancellation" callback, and then these returns is replaced by a ``cancel()`` which returns, but runs the registered callback first. To allow a module to roll back changes or remove spawned stuff?

Reply:
Actually went for deleting in 6b537f1 after all, since it's the easiest way and also doesn't couple `GME_AmbientFlybyComponent` with the module when it comes to the clean-up.